koku/masu/api/update_cost_model_costs.py

```diff
@@ -53,6 +54,11 @@ def update_cost_model_costs(request):
 
     LOG.info("Calling update_cost_model_costs async task.")
 
-    async_result = cost_task.delay(schema_name, provider_uuid, start_date, end_date)
+    try:
+        provider = Provider.objects.get(uuid=provider_uuid)
+    except Provider.DoesNotExist:
+        return Response({"Error": "Provider does not exist."}, status=status.HTTP_400_BAD_REQUEST)
+    else:
+        async_result = cost_task.delay(schema_name, provider_uuid, start_date, end_date, provider.type)
```

Comment:
I need to think on the implications of making this asynchronous.  The main concern I have is that the API is synchronous but if the work to update the report data is asynchronous, it might impact QE tests since from they will not know when the actual processing is complete.

Reply:
The only thing that was added here was the try/except. This was already an async call.